a00_qpip/plugin.py

```diff
@@ -170,6 +175,8 @@ def check_deps_and_prompt_install(self, additional_plugins=[], force_gui=False):
             dialog = MainDialog(
                 libs.values(), self._check_on_startup(), self._check_on_install()
             )
+            if self.test_mode:
+                return dialog.reqs_to_install
```

Comment:
Could we extract this to a separate method and test this one instead of having this `test_mode` here?

Reply:
Hmm, I'm not sure what you suggest, I wanted to run as much of the function as possible but "break out of it" before it hits the `dialog._exec()`, I could return the dialog to test for more stuff as well, but it sounds like you had something else on your mind?